api/src/AsyncRunner.php

```diff
@@ -64,17 +64,17 @@ public function cleanUp() {
 	}
 
 	/**
-	 * 
-	 * @throws AsyncRunnerException
+	 * Waits for up to 5s for the runner to complete
+	 * @return boolean	True if complete, otherwise not complete yet
 	 */
 	public function synchronize() {
```

Comment:
how about we rename synchronize() to be isComplete() or isFinished() ?  The previous implementation would ensure synchronization or throw an exception (or at least that was the intention).  The new functionality just waits up to 5 seconds for the job to complete before returning false.
```suggestion
	public function isComplete() {
```
We would need to change the references if we rename.

Reply:
Renamed it to `waitForIsComplete`. `isComplete` already exists. It's what this uses while it waits.